src/main/java/com/google/sps/servlets/RecipeServlet.java

```diff
@@ -42,18 +42,30 @@
 @WebServlet("/api/post")
 public class RecipeServlet extends HttpServlet {
   private Gson gson = new Gson();
+  private Date date = new Date();
+  private DBInterface db;
+
+  public RecipeServlet(DBInterface mockInterface) {
```

Comment:
is this constructor only used for mocks?

Reply:
Yep as far as I know. In normal startup situations a default constructor is called (which I need to add)